app/models/manageiq/providers/embedded_ansible/automation_manager/playbook.rb

```diff
@@ -1,48 +1,50 @@
 
-  def run(options, userid = nil)
-    options[:playbook_id] = id
-    options[:userid]      = userid || 'system'
-    options[:name]        = "Playbook: #{name}"
-    miq_job = ManageIQ::Providers::EmbeddedAnsible::AutomationManager::PlaybookRunner.create_job(options)
-    miq_job.signal(:start)
-    miq_job.miq_task.id
-  end
-
-  # return provider raw object
-  def raw_create_job_template(options)
-    job_template_klass = ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ConfigurationScript
-    jt_options = build_parameter_list(options)
-    _log.info("Creating job template with options:")
-    $log.log_hashes(jt_options)
-    job_template_klass.raw_create_in_provider(manager, jt_options)
-  end
+  DEFAULT_EXECUTION_TTL = 100.minutes # automate state machine aborts after 100 retries at a minite interval
 
   def self.display_name(number = 1)
     n_('Playbook (Embedded Ansible)', 'Playbooks (Embedded Ansible)', number)
   end
 
-  private
+  def run(vars = {})
+    workflow = ManageIQ::Providers::AnsiblePlaybookWorkflow
 
-  def build_parameter_list(options)
-    params = {
-      :name                     => options[:template_name] || "#{name}_#{SecureRandom.uuid}",
-      :description              => options[:description] || "Created on #{Time.zone.now}",
-      :project                  => configuration_script_source.manager_ref,
-      :playbook                 => name,
-      :become_enabled           => options[:become_enabled].present?,
-      :verbosity                => options[:verbosity].presence || 0,
-      :limit                    => options[:limit],
-      :inventory                => options[:inventory] || manager.provider.default_inventory,
-      :extra_vars               => options[:extra_vars].try(:to_json)
+    extra_vars = build_extra_vars(vars[:extra_vars])
+
+    playbook_vars = {
+      :configuration_script_source_id => configuration_script_source_id,
+      :playbook_relative_path         => name
     }
 
-    %i(credential vault_credential cloud_credential network_credential).each do |credential|
-      cred_sym = "#{credential}_id".to_sym
-      params[credential] = Authentication.find(options[cred_sym]).native_ref if options[cred_sym].present?
+    credentials = collect_credentials(vars)
+
+    kwargs = {:become_enabled => vars[:become_enabled]}
+    kwargs[:timeout]   = vars[:execution_ttl].present? ? vars[:execution_ttl].to_i.minutes : DEFAULT_EXECUTION_TTL
+    kwargs[:verbosity] = vars[:verbosity].to_i if vars[:verbosity].present?
+
+    workflow.create_job({}, extra_vars, playbook_vars, vars[:hosts], credentials, kwargs).tap do |job|
+      job.signal(:start)
     end
+  end
+
+  private
+
+  def build_extra_vars(external = {})
+    extra_vars = (external || {}).each_with_object({}) do |(k, v), hash|
+      match_data = v.kind_of?(String) && /password::/.match(v)
+      hash[k] = match_data ? ManageIQ::Password.decrypt(v.gsub(/password::/, '')) : v
+    end
+
+    extra_vars
```

Comment:
No need to set a temporary variable and return it because it would be returned anyway.

Reply:
To be fair, this was a copy-paste from what was done in `EmbeddedAnsible::AutomationManager::ConfigurationScript` previously.